paywall/src/middlewares/web3Middleware.js

```diff
@@ -134,41 +156,25 @@ export default function web3Middleware({ getState, dispatch }) {
         // So that pending API calls do not interract with our "new" state.
         web3Service.refreshAccountBalance(action.account)
 
-        const {
-          router: {
-            location: { pathname },
-          },
-        } = getState()
-
-        const { lockAddress } = lockRoute(pathname)
         if (lockAddress) {
           web3Service.getKeyByLockForOwner(lockAddress, action.account.address)
```

Comment:
Combine this with line 146 above?

Reply:
I tried, but these are actually quite different. 146 uses the URL lock, but this line uses the lock from the action, which may be completely different. I'll add a comment